projects/common/src/telemetry/user-telemetry-impl.service.ts

```diff
@@ -29,31 +30,32 @@ export class UserTelemetryImplService extends UserTelemetryService {
   }
 
   public initialize(): void {
-    this.telemetryProviders.forEach(provider => provider.telemetryProvider.initialize(provider.initConfig));
+    this.allTelemetryProviders.forEach(provider => provider.telemetryProvider.initialize(provider.initConfig));
+    this.initializedTelemetryProviders = [...this.allTelemetryProviders];
```

Comment:
what are we trying to solve here? were we calling before initializing? That's the only behavior that I see that would change here. Wouldn't that imply a misconfig though?

Reply:
Currently, we don't 'initialize' telemetry when FF is disabled or when user is an internal user. But the track directive still pushes click events on the service. Our E2E tests fails when it sees any console error. In this case, the third party lib throws an error since we tried to send an 'event' without initializing the library.

Reply:
The other alternative is to put a init check in every method that will run every time the track event methods are called. We can avoid that by keeping a separate initialized list

Reply:
Still, a perf hit. But an ideal solution would be to load the entire module only if the feature flag is enabled.